### DISCUSSION TITLE
Annoyance - My last session's result is auto-populated

### DISCUSSION THREAD

**Discussion by revionics-ryan-felder** [elementary/calculator#156] [General]
When I close the calculator, I'm done with the results from that session.

When I open the calculator, I would like to begin using it without having to clean up my state from last time.

I also cannot see an easy way to clear the calculator without using the mouse.

Here's what happens to me 99% of the time when I use the elementary calculator...

Super+space calc \<enter\> 140*4 \<enter\> .. 5602? Ah hell...

If I intentionally click the X to close the calculator, I'm done with it. 
I don't want my previous session's results anymore.

**Comment by cassidyjames** [comment]
Hm, there's some decent feedback here. 

>When I close the calculator, I'm done with the results from that session. When I open the calculator, I would like to begin using it without having to clean up my state from last time.

For context, apps in elementary OS save their state and restore where you left off; Files, Terminal, Calculator, the web browser, etc. all behave this way. You can disable this behavior system-wide by turning off “History” in _System Settings_ → _Security & Privacy_. Changing this behavior in Calculator by default might be nice for you, but then anyone who relies on that state saving is going to be even more annoyed. 😉 

>I also cannot see an easy way to clear the calculator without using the mouse.

Tapping <kbd>Esc</kbd> clears the entry, and that shortcut is listed in the tooltip when hovering the button.

To address your workflow, though, we could highlight the text when Calculator opens. This way typing anything new would replace the previous results, but someone wanting to see the previous results would get an opportunity to do so. I've filed this as https://github.com/elementary/calculator/issues/157

**Reply by revionics-ryan-felder** [reply]
Thanks for the info! I'm surprised I never tried the Esc key to clear the field.
I'm following the discussion on #157 now and look forward to seeing how it develops.
